src/docs/sphinx/dev_guide/functional.rst

```diff
@@ -141,24 +189,113 @@ Putting these snippets together without the verbose comments, we have (note: the
 
 .. code-block:: cpp
 
-   using test = H1<order, dim>;
-   using trial = H1<order, dim>;
-   Functional< test(trial) > residual(test_fes, trial_fes);
+    using test = H1<order, dim>;
+    using trial = H1<order, dim>;
+    Functional< test(trial) > residual(test_fes, trial_fes);
 
-   // note: the first two AddVolumeIntegral calls can be fused
-   // into one, provided they share the same domain of integration
-   residual.AddVolumeIntegral([](auto x, auto disp){
-     auto [u, grad_u] = disp;
-     return std::tuple{b(x), material_model(grad_u))};
-   }, mesh);
+    // note: the first two AddVolumeIntegral calls can be fused
+    // into one, provided they share the same domain of integration
+    residual.AddVolumeIntegral(
+      DependsOn<0>{}, // depends on the displacement field
+      [](auto x, auto disp){
+        auto [u, grad_u] = disp;
+        return serac::tuple{b(x), material_model(grad_u)};
+      }, 
+      mesh
+    );
 
-   residual.AddSurfaceIntegral([](auto x, auto disp /* unused */){ return traction(x); }, surface_mesh);
+    residual.AddSurfaceIntegral([](auto x, auto disp /* unused */){ return traction(x); }, surface_mesh);
 
-   auto r = residual(displacements);
+    auto r = residual(displacements);
 
 So, in only a few lines of code, we can create optimized, custom finite
 element kernels!
 
+Quantities of Interest
+----------------------
+
+``Functional`` can also be used to represent scalar-valued integral expressions. These can be used
+to represent objective functions, constraints, or other "quantities of interest". To make a ``Functional``
+with a scalar-valued output, use ``double`` as the test space in its function signature:
+
+.. code-block:: cpp
+
+    using test = H1<order, dim>;
```

Comment:
Are the `trial` and `test` symbols backwards here? Since the signature is `test(trial)`, I would have expected `test` to be `double` in this case.

Reply:
good catch!